Makefile

```diff
@@ -59,17 +59,25 @@ non_windows_platforms := darwin_amd64 darwin_arm64 linux_amd64 linux_arm64
 # TODO: arm64 on Windows https://github.com/envoyproxy/envoy/issues/17572
 windows_platforms := windows_amd64
 
-$(build_dir)/func-e_%/func-e: main.go
-	$(call go-build, $@, $^)
+# TODO: To have expression that understand globbing for any depth.
+gocodes := [!_test].go
+subdirs := $(wildcard */)
+sources := $(wildcard \
+	$(addsuffix *$(gocodes),$(subdirs)) \
+	$(addsuffix */*$(gocodes),$(subdirs)) \
+	$(addsuffix */*/*$(gocodes),$(subdirs)))
+
+$(build_dir)/func-e_%/func-e: main.go $(sources)
+	$(call go-build, $@, main.go)
```

Comment:
I like the caret (hat?) but seems like we need to change it to `main.go`.

Reply:
According to my constant reference Managing Projects with GNU Make", `$<` should get the first dependency. Would that work?

Reply:
Ah, yeah! Thanks, @codefromthecrypt!